app/Http/ViewComposers/BannerComposer.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace App\Http\ViewComposers;
+
+
+use App\SpecialOffer;
+use Carbon\Carbon;
+use Illuminate\Contracts\View\View;
+use Illuminate\Support\Facades\Auth;
+
+class BannerComposer
+{
+    public function compose(View $view)
+    {
+        $user = Auth::user();
+        $offers = SpecialOffer::all();
```

Comment:
adminui taip pat turėtų būti aktualūs tik vis dar galiojantys special offeriai. ir turėtų būti vykdoma tik viena užklausa, priklausomai nuo to ar admin ar ne. pabandyk pritaikyt panašų principą kaip darei su filtru